### PR TITLE
Linting

### LINKED ISSUE #171: Reboot cf-units
To-do checklist:
- [x] Enable GH discussions
  - [x] Create Releases GH discussions category
  - [x] Create `3.0` release discussion 
- [x] Add Issue templates
- [x] Add PR template
- [x] Create `3.0` release milestone
- [x] Create `3.0` project board
- [x] Update labels
- [x] Configure repo settings
  - [x] Allow squash merging only
  - [x] Require conversation resolution before merging
  - [x] Require linear history
- [x] Migrate CI from `travis-ci` to `cirrus-ci`
- [x] Adopt `setuptools-scm` and drop `versioneer`
- [x] Adopt `pre-commit`
  - [x] `black`
  - [x] `flake8`
  - [x] `isort`
- [x] Migrate `setup.py` to `setup.cfg`
- [x] Rationalise package dependencies for `conda` and PyPI
- [x] Update `README.md` badges
- [x] Remove `six` dependency
- [x] Simpler licence headers (e.g. SciTools/iris#3489)

### PR DESCRIPTION
**Pull request by trexfeathers**
### Description

Closes #171.

Adds the 'standard' SciTools linters, and tidies some other small things.

### CODE REVIEW

**Comment by trexfeathers** [comment]
The coverage reduction I think is to be expected, given the lines that Black tends to add. Depending on the reviewer's preference, I am happy for us to:

a. Temporarily alter the coverage settings to be tolerant of this shift.
b. Merge with a CI failure.

Let me know 🙂

**Comment by bjlittle** [comment]
@trexfeathers Since there might be a bit of fluctuation given that we're moving stuff around and linting et al, I'd suggest making the codecov regression suitably large to accommodate changes, then once the dust settles, bump it back to `0.5%`

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    -   id: debug-statements
+        # Don't commit to main branch.
+    -   id: no-commit-to-branch
+-   repo: https://github.com/psf/black
+    rev: '21.6b0'
+    hooks:
+    -   id: black
+        # Force black to run on whole repo, using settings from pyproject.toml
+        pass_filenames: false
+        args: [--config=./pyproject.toml, .]
```
Comment (bjlittle) [inline .pre-commit-config.yaml]: @trexfeathers This `pre-commit` hook for `black` should now only perform a `--check`, see https://github.com/SciTools/iris/pull/4181

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    hooks:
+        # Run flake8.
+    -   id: flake8
+        args: [--config=./setup.cfg]
+-   repo: https://github.com/pycqa/isort
+    rev: '5.8.0'
+    hooks:
+    -   id: isort
+        name: isort
+        args: ["--profile", "black", "--filter-files"]
```
Comment (bjlittle) [inline .pre-commit-config.yaml]: @trexfeathers This `pre-commit` hook for `isort` should now only perform a `--check`, see https://github.com/SciTools/iris/pull/4181

**Review by bjlittle — CHANGES_REQUESTED** [review]
@trexfeathers Looking good... just a couple of comments to service :+1:

File: .pre-commit-config.yaml (hunk 1 above)
Comment (trexfeathers) [reply .pre-commit-config.yaml]: 7555e8ff47d418081c4f8846bea6745f7c271716

File: .pre-commit-config.yaml (hunk 2 above)
Comment (trexfeathers) [reply .pre-commit-config.yaml]: 7555e8ff47d418081c4f8846bea6745f7c271716

**Review by bjlittle — APPROVED** [review]

**Comment by trexfeathers** [comment]
Thanks @bjlittle !

**Pull request merged**